src/main/java/uk/gov/companieshouse/company/metrics/service/DeletedChargesClient.java

```diff
@@ -58,35 +56,45 @@ public void postMetrics(String companyNumber,
-                            + "charges recalculate request", ex);
-                } else if (ex.getStatusCode() == HttpStatus.NOT_FOUND.value()) {
-                    logger.info("HTTP 404 Not Found returned; "
-                            + "company does not exist");
-                } else {
-                    logger.error(String.format("Changed charges client error returned with "
-                                    + "status code: [%s] when processing recalculate request",
-                            ex.getStatusCode()));
-                    throw new NonRetryableErrorException("UpsertClient error returned when "
-                            + "processing charges recalculate request", ex);
-                }
+                handleApiError(companyNumber, ex);
             } catch (IllegalArgumentException ex) {
-                logger.error("Illegal argument exception caught when handling API response");
-                throw new RetryableErrorException("Server error returned when processing "
-                        + "charges recalculate request", ex);
+                handleIllegalArguementError(companyNumber, ex);
             } catch (URIValidationException ex) {
-                logger.error("Invalid companyNumber specified when handling API request");
-                throw new NonRetryableErrorException("Invalid companyNumber specified", ex);
+                handleURIValidationError(companyNumber, ex);
             }
         } else {
-            throw new RetryableErrorException("Charge details found when should have been deleted");
+            throw new RetryableErrorException("Charge details found for [%s] when " +
+                    "should have been deleted " + companyNumber);
+        }
+    }
+
+    private void handleURIValidationError(String companyNumber, URIValidationException ex) {
+        String message = String.format(FAILED_MSG, companyNumber);
+        logger.error(message);
+        throw new NonRetryableErrorException(message, ex);
+    }
+
+    private void handleIllegalArguementError(String companyNumber, IllegalArgumentException ex) {
+        String message = String.format(FAILED_MSG, companyNumber);
+        logger.error(message);
+        throw new RetryableErrorException(message, ex);
+    }
+
+    private void handleApiError(String companyNumber, ApiErrorResponseException ex) {
+        String message = String.format(ERROR_MSG, ex.getStatusCode(), companyNumber);
+        if (HttpStatus.valueOf(ex.getStatusCode()).is5xxServerError()) {
+            logger.info(message);
+            throw new RetryableErrorException(message, ex);
+        } else if (ex.getStatusCode() == HttpStatus.NOT_FOUND.value()) {
+            logger.info(message);
+        } else {
+            logger.error(message);
+            throw new NonRetryableErrorException(message, ex);
         }
     }
 
     private boolean isChargeAlreadyDeleted(String resourceUri, String contextId) {
+        //include in tech debt ticket.
```

Comment:
As above, this probably should not be a commit in the final commit. It might still be there in a few years but nobody will understand the context nor have the courage to delete it